app.js

```diff
@@ -0,0 +1,188 @@
+  // you do not want to store your password in plain-text here.
+  const uri =
+    "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority";
+
+  // The MongoClient is the object that references the connection to our
+  // datastore (Atlas, for example)
+  const client = new MongoClient(uri);
+
+  // The connect() method does not attempt a connection; instead it instructs
+  // the driver to connect using the settings provided when a connection
+  // is required.
+  await client.connect();
+
+  // Provide the name of the database and collection you want to use.
+  // If the database and/or collection do not exist, the driver and Atlas
+  // will create them automatically when you first write data.
+  const dbName = "myDatabase";
+  const collectionName = "recipes";
+
+  // Create references to the database and collection in order to run
+  // operations on them.
+  const database = client.db(dbName);
+  const collection = database.collection(collectionName);
+
+  /*
+   *  *** INSERT DOCUMENTS ***
+   *
+   * You can insert individual documents using collection.insert().
+   * In this example, we're going to create four documents and then
+   * insert them all in one call with collection.insertMany().
+   */
+
+  const recipes = [
+    {
+      name: "elotes",
+      ingredients: [
+        "corn",
+        "mayonnaise",
+        "cotija cheese",
+        "sour cream",
+        "lime",
+      ],
+      prepTimeInMinutes: 35,
+    },
+    {
+      name: "loco moco",
+      ingredients: [
+        "ground beef",
+        "butter",
+        "onion",
+        "egg",
+        "bread bun",
+        "mushrooms",
+      ],
+      prepTimeInMinutes: 54,
+    },
+    {
+      name: "patatas bravas",
+      ingredients: [
+        "potato",
+        "tomato",
+        "olive oil",
+        "onion",
+        "garlic",
+        "paprika",
+      ],
+      prepTimeInMinutes: 80,
+    },
+    {
+      name: "fried rice",
+      ingredients: [
+        "rice",
+        "soy sauce",
+        "egg",
+        "onion",
+        "pea",
+        "carrot",
+        "sesame oil",
+      ],
+      prepTimeInMinutes: 40,
+    },
+  ];
+
+  collection
+    .insertMany(recipes)
+    .then(result =>
+      console.log(`${result.insertedCount} documents successfully inserted.`),
+    )
+    .catch(err =>
+      console.error(
+        `Something went wrong trying to insert the new documents: ${err}`,
+      ),
+    );
+
+  /*
+   * *** FIND DOCUMENTS ***
+   *
+   * Now that we have data in Atlas, we can read it. To retrieve all of
+   * the data in a collection, we call Find() with an empty filter.
+   * The Builders class is very helpful when building complex
+   * filters, and is used here to show its most basic use.
+   */
+
+  const findQuery = { prepTimeInMinutes: { $lt: 45 } };
+
+  try {
+    const cursor = await collection.find(findQuery).sort({ name: 1 });
+    await cursor.forEach(recipe =>
```

Comment:
I *think* `forEach` is synchronous, but lemme know if I'm looking at [outdated docs](http://mongodb.github.io/node-mongodb-native/3.5/api/AggregationCursor.html#forEach).
```suggestion
    cursor.forEach(recipe =>
```

Reply:
Oops, good catch!

Reply:
Actually, the docs contain a typo. It should read:
"Promise if no callback passed"
not "no callback passed"

Therefore, we want `await` here unless you don't care when these print which could be during the next operation.